### ISSUE TITLE
AutoComplete Typescript Error

### ISSUE THREAD

**Issue by Knacktus** [ant-design/ant-design#5170]
First of all, thanks for creating and maintaining such a great framework. 

So far all was working fine, but now I've possibly found an issue.

AntD Version: 2.7.4

Using the AutoComplete with Typescript like this:

```
                                    <AutoComplete
                                        dataSource={fulltext}
                                        style={{ width: 400 }}
                                        placeholder="Volltext Suche"
                                        onSelect={this.onFullTextSelected}
                                        onChange={this.onFullTextChanged}
                                    >
                                    <Input.Search />
                                    </AutoComplete>
```

gives the Typescript Error:

`[ts] Property 'children' is missing in type 'IntrinsicAttributes & IntrinsicClassAttributes<AutoComplete> & Readonly<{ children?: ReactNode; }...'.`

When I modify the typings as follows the Error is gone.

**Old:**

```
export interface AutoCompleteProps extends AbstractSelectProps {
    size?: 'large' | 'small' | 'default';
    className?: string;
    notFoundContent?: Element;
    dataSource: DataSourceItemType[];
    defaultValue?: string | Array<any> | SelectedValue | Array<SelectedValue>;
    value?: string | Array<any> | SelectedValue | Array<SelectedValue>;
    onChange?: (value: string | Array<any> | SelectedValue | Array<SelectedValue>) => void;
    onSelect?: (value: string | Array<any> | SelectedValue | Array<SelectedValue>, option: Object) => any;
    disabled?: boolean;
    children: ValidInputElement | React.ReactElement<OptionProps> | Array<React.ReactElement<OptionProps>>;
}
```

**New (I made the children property optional):**
```
...
children?: ValidInputElement | React.ReactElement<OptionProps> | Array<React.ReactElement<OptionProps>>;
...
```

Is this an error in the definition or do I use the AutoComplete component wrong?

Cheers,

Jan

**Comment by paranoidjk** [comment]
- you should provide you antd version first.
- have you tried [this demo](https://ant.design/components/auto-complete/#components-auto-complete-demo-uncertain-category) ?  it use `Input` as children works fine

**Comment by Knacktus** [comment]
I'm using version 2.7.4 (also updated my initial post). The examples worked fine using them with Javascript. Now I've moved to Typescript. The Error is the same when I run the demo. It seems a Typescript issue only.

**Comment by paranoidjk** [comment]
The reason is Typescript recently change ...

https://github.com/Microsoft/TypeScript/issues/6471

cc @afc163

**Comment by benjycui** [comment]
@Knacktus just PR to update this definition.

**Comment by paranoidjk** [comment]
@afc163 maybe there is more same problem ？

https://github.com/ant-design/ant-design/search?l=TypeScript&q=children&type=Code&utf8=%E2%9C%93

https://github.com/ant-design/ant-design/blob/8b539c497100a2a135924d2287ddcea5945727b4/components/form/FormItem.tsx#L53

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/blob/8b539c497100a2a135924d2287ddcea5945727b4/components/form/FormItem.tsx#L53

It is not typescript definition.

**Comment by benjycui** [comment]
It seems that this problem will be solved in TypeScript@2.3, trace https://github.com/Microsoft/TypeScript/issues/13618
